src/test.js

```diff
@@ -550,6 +550,13 @@ function test( testName, expected, callback, async ) {
 	if ( arguments.length === 2 ) {
 		callback = expected;
 		expected = null;
+	} else if ( arguments.length === 3 && typeof expected !== "function" ) {
+		if ( global.console && global.console.warn ) {
+			global.console.warn(
+				"'expected' argument of QUnit.test is deprecated. " +
+				"Call the expect() function explicitly instead."
```

Comment:
> Did you pushed the commit with it? I would love to see it.

No, I was waiting for your feedback for my other comments. I will update the PR in a day or two.

Thanks!